packages/shared/src/UI/components/PublicWalletSetting/index.tsx

```diff
@@ -0,0 +1,215 @@
+import { Icons } from '@masknet/icons'
+import { LoadingBase, makeStyles, useCustomSnackbar } from '@masknet/theme'
+import { Typography, Box } from '@mui/material'
+import { memo, useCallback, useMemo, useState } from 'react'
+import {
+    BindingProof,
+    CrossIsolationMessages,
+    ECKeyIdentifier,
+    EMPTY_LIST,
+    NextIDPlatform,
+    PopupRoutes,
+} from '@masknet/shared-base'
+import { useHiddenAddressSetting, useWeb3State } from '@masknet/plugin-infra/web3'
+import { PluginID as MaskPluginID } from '@masknet/plugin-infra'
+import { WalletSettingCard } from '@masknet/shared'
+import { useAsyncFn, useUpdateEffect } from 'react-use'
+
+import { differenceWith } from 'lodash-unified'
+import { isSameAddress } from '@masknet/web3-shared-base'
+import { useSharedI18N } from '../../../locales'
+import { SettingActions } from './SettingActions.js'
+
+export type PublicWalletSettingType = {
+    hiddenAddresses?: string[]
+    defaultAddress?: string
+}
+
+const useStyles = makeStyles()((theme) => ({
+    container: {
+        padding: 16,
+        minHeight: 424,
+        display: 'flex',
+        flexDirection: 'column',
+    },
+    alert: {
+        padding: theme.spacing(1.5),
+        background: theme.palette.maskColor.bg,
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        borderRadius: 4,
+    },
+    alertTitle: {
+        fontSize: 14,
+        lineHeight: '18px',
+    },
+    placeholder: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        flex: 1,
+        rowGap: 12,
+    },
+    placeholderText: {
+        color: theme.palette.maskColor.second,
+        fontSize: 14,
+        lineHeight: '18px',
+    },
+    placeholderIcon: {
+        color: theme.palette.maskColor.third,
+    },
+    content: {
+        marginTop: 12,
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: 12,
+        '& > *': {
+            flex: 1,
+            maxWidth: 'calc(50% - 22px)',
+        },
+    },
+}))
+
+interface PublicWalletSettingProps {
+    onClose: () => void
+    onOpenPopup: (route?: PopupRoutes, params?: Record<string, any>) => void
+    bindingWallets?: BindingProof[]
+    currentPersona?: ECKeyIdentifier
+    pluginId: MaskPluginID
+}
+
+export const PublicWalletSetting = memo<PublicWalletSettingProps>(
```

Comment:
```suggestion
export const PublicWalletSettings = memo<PublicWalletSettingProps>(
```

Reply:
I think should remove 's', this component is only work for someone plugin.